### ISSUE TITLE
Do not include `#[test]` functions

### ISSUE THREAD

**Issue by snaulX** [Cysharp/csbindgen#58]
I create tests for bindings in the same file with bindings themselves (or else it gets me a lot of troubles with visibility). And these test functions are includuded in bindings for C# so maybe don't include functions with `#[test]` attribute? Or maybe even include only functions with `#[no_mangle]` attribute? I can make pull request if this feature is needed.

**Comment by neuecc** [comment]
The only current extraction is the extern method.
Is this is almost identical to no_mangle?

**Comment by snaulX** [comment]
As I know, `no_mangle` disables Rust name mangling for the function, so it will have a normal name without internal compiler information. So we remove functions with '"strange names" from bindings as we check it on `no_mangle`.
Also I think that there's need check of pub visibility because else function just  will not be showed in dll as exported.

**Comment by hadashiA** [comment]
Thanks for the suggenstion. I think you are right that we should check for `no_mangle`.

- However, for `#[test]`, it should not be covered without extern.
- Also, the `pub` access modifier is also a visibility on Rust and should not be covered by ffi.

I have made a fixing to your point #60. I also commented here. I hope you will check it. https://github.com/Cysharp/csbindgen/pull/60#issuecomment-1878194699
